### PR TITLE
Add --find0 and --list0 flags

### PR DESCRIPTION
**Pull request by sdavids**
#1096

### CODE REVIEW

**Comment by sdavids** [comment]
One thing I noticed but I did not want to create a new issue or pollute this PR:

https://github.com/mvdan/sh/blob/57b7613429b5ac2918c3218f321107f567ba2165/_js/go.mod#L3

should it not be the same as:

https://github.com/mvdan/sh/blob/57b7613429b5ac2918c3218f321107f567ba2165/go.mod#L3

File: cmd/shfmt/main.go [hunk 1]
```diff
@@ -420,9 +447,13 @@ func formatPath(path string, checkShebang bool) error {
 		}
 		readBuf.Write(copyBuf[:n])
 	}
-	if find.val {
+	if find.val == "true" {
 		fmt.Println(path)
 		return nil
+	} else if find.val == "0" {
```
Comment (mvdan) [inline cmd/shfmt/main.go:453]: use a switch on find.val

File: cmd/shfmt/main.go [hunk 2]
```diff
@@ -484,8 +515,11 @@ func formatBytes(src []byte, path string, fileLang syntax.LangVariant) error {
 	printer.Print(&writeBuf, node)
 	res := writeBuf.Bytes()
 	if !bytes.Equal(src, res) {
-		if list.val {
+		if list.val == "true" {
 			fmt.Println(path)
+		} else if list.val == "0" {
```
Comment (mvdan) [inline cmd/shfmt/main.go:520]: use a switch on list.val

**Review by mvdan — APPROVED** [review]
thanks, just two nits

**Review by mvdan — APPROVED** [review]
Thanks!

**Comment by mvdan** [comment]
FYI force-pushed here to resolve the conflict, squash the commits, and write a slightly longer commit message.

**Pull request merged**